test/acceptance.hs

```diff
@@ -399,11 +371,35 @@ sendAuthorization uri user_m fields = do
             redirect <- case lookup hLocation hs of
                 Nothing -> throwError "No Location header in redirect"
                 Just x' -> return x'
-            case parseURI $ BC.unpack redirect of
-                Nothing -> throwError $ "Invalid Location header in redirect: " <> show redirect
-                Just x -> return x
+            case UB.parseURI UB.strictURIParserOptions redirect of
+                Left e -> throwError $ "Invalid Location header in redirect: " <> show (redirect, e)
+                Right x -> return x
         Right _ -> throwError "No redirect"
 
+-- |
```

Comment:
I like that you made an effort, but perhaps there should be more content to this comment.